src/connector/src/sink/snowflake.rs

```diff
@@ -0,0 +1,279 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use anyhow::anyhow;
+use async_trait::async_trait;
+use risingwave_common::array::{Op, StreamChunk};
+use risingwave_common::buffer::Bitmap;
+use risingwave_common::catalog::Schema;
+use serde::Deserialize;
+use serde_json::Value;
+use serde_with::serde_as;
+use with_options::WithOptions;
+
+use super::encoder::{
+    JsonEncoder, RowEncoder, TimeHandlingMode, TimestampHandlingMode, TimestamptzHandlingMode,
+};
+use super::snowflake_connector::{SnowflakeHttpClient, SnowflakeS3Client};
+use super::writer::LogSinkerOf;
+use super::{SinkError, SinkParam};
+use crate::sink::writer::SinkWriterExt;
+use crate::sink::{DummySinkCommitCoordinator, Result, Sink, SinkWriter, SinkWriterParam};
+
+pub const SNOWFLAKE_SINK: &str = "snowflake";
+const MAX_BATCH_ROW_NUM: u32 = 1000;
```

Comment:
Can we let the user configure this item

Reply:
Presumably this could be a configurable option, if the current sink logic overall looks good.

Reply:
Had better use file size as threshold.

Reply:
> Had better use file size as threshold.

What does "file size" mean?
Do we have the ability to know the exact size of the batch row(s) during runtime? (we can base on the previous S3 statistics, but this will 1. involve extra query request to S3, which will downgrade the performance and make the logic more complicated 2. the size may not be exact the same)

Reply:
cc @fuyufjh.